Mathlib/Data/Finset/Basic.lean

```diff
@@ -2579,6 +2580,7 @@ theorem nonempty_range_iff : (range n).Nonempty ↔ n ≠ 0 :=
 theorem range_eq_empty_iff : range n = ∅ ↔ n = 0 := by
   rw [← not_nonempty_iff_eq_empty, nonempty_range_iff, not_not]
 
+@[aesop safe apply (rule_sets := [finsetNonempty])]
 theorem nonempty_range_succ : (range <| n + 1).Nonempty :=
```

Comment:
That's less syntactically general than the lemma you're unaesoping. Can you fix it?

Reply:
Unfortunately the story here is a little more complex than "syntactically general": the version that you suggest would not prove `0 < (range (n + 1)).card` by `positivity`. So this version genuinely is needed, for the current setup of `positivity`.

Reply:
Hmmm, annoying